### PR TITLE
ACA-5356: Add VaultDatabaseDynamicRoles client methods and unit tests (+refactor database class)

### PR DESCRIPTION
**Pull request by mjohns91**
## Summary

Implements client methods for managing database dynamic roles in the Vault Database Secrets Engine, and refactors all database-related classes into a dedicated module for better code organization.

### Database Dynamic Roles Feature
Add `VaultDatabaseDynamicRoles` class with CRUD operations:
- `list_dynamic_roles()` - List all dynamic role names
- `read_dynamic_role()` - Read dynamic role configuration
- `create_or_update_dynamic_role()` - Create/update with validation
- `delete_dynamic_role()` - Delete a dynamic role

Update `Database` container class to include `dynamic_roles` client for managing dynamic database roles

All methods follow the same pattern as `VaultDatabaseStaticRoles` and include proper error handling, type validation, and support for custom mount paths.

### Code Organization Refactoring
Refactor database classes into dedicated `vault_database.py` module:
- Created new `plugins/module_utils/vault_database.py` (484 lines)
- Moved `VaultDatabaseConnection`, `VaultDatabaseStaticRoles`, `VaultDatabaseDynamicRoles`, and `Database` classes
- Reduced `vault_client.py` from 1052 to 622 lines (~40% reduction)
- Updated all imports across modules and tests to use `vault_database`

### Testing
- Add 19 comprehensive unit tests for dynamic roles covering all operations and edge cases
- Update Database container tests to verify `dynamic_roles` initialization
- All 65 database-related unit tests passing
- All linters passing (black, flake8, isort)

### CODE REVIEW

File: plugins/module_utils/vault_client.py [hunk 1]
```diff
@@ -410,32 +410,167 @@ def get_static_role_credentials(self, name: str, read_snapshot_id: Optional[str]
...
+        Initializes the Database dynamic roles client.
+
+        Args:
+            client (VaultClient): An authenticated instance of the main VaultClient.
+            mount_path (str): The mount path of the database secrets engine. Defaults to "database".
+        """
+        self._client = client
+        self._mount_path = (mount_path or "database").strip().strip("/")
+
+    def list_dynamic_roles(self) -> list:
```
Comment (alinabuzachis) [inline plugins/module_utils/vault_client.py]:
```suggestion
    def list_dynamic_roles(self) -> List[str]:
```

**Review by hdefazio — APPROVED** [review]

**Review by abikouo — COMMENTED** [review]
Great work on this overall! I noticed that the three Database* classes share an identical constructor. To keep the code DRY (Don't Repeat Yourself), please refactor these to inherit from a common parent class that handles the shared initialization.

```
class DatabaseParent(object):
  def __init__(self, client, mount_path):
      self.client = client
      self.mount_path = (mount_path or "database").strip("").strip("/")


class DatabaseConnection(DatabaseParent):
...
```

As vault_client.py continues to grow, it might be a good time to refactor. Consider moving the three database-related classes into a dedicated vault_database_client.py file to improve maintainability.

**Review by abikouo — CHANGES_REQUESTED** [review]
@mjohns91 the refactoring is OK Do you mind adding the corresponding `DatabaseParent`?

**Review by abikouo — APPROVED** [review]

**Review by hdefazio — APPROVED** [review]

**Review by beeankha — APPROVED** [review]
